test/map.carp

```diff
@@ -78,12 +83,17 @@
     )
     (assert-equal test
                   false
-                  (Set.empty? &(Set.put (Set.create) "1"))
+                  (Set.empty? &(Set.put &(Set.create) "1"))
                   "empty? works"
     )
     (assert-equal test
                   true
-                  (Set.empty? &(Set.remove (Set.put (Set.create) "1") "1"))
+                  (Set.empty? &(Set.remove (Set.put &(Set.create) "1") "1"))
                   "remove works"
     )
+    (assert-equal test
+                  "{ hi bye }"
```

Comment:
Shouldn't the strings have quotes here? (like with `str` for Array)

Reply:
Depends. `str` of `String` doesn’t do that, `prn` does IIRC.

Reply:
Yeah, String's the reason there are two different functions. It's certainly cleaner looking to have it like this but also confusing since you can't really see that those are strings and not variable names. Since structs and arrays do the other thing I think map should behave the same way.

Reply:
Makes sense!